### PR TITLE
refactor: journey refactor notes

### PR DESCRIPTION
**Pull request by bryn-pins**
## Description of change
 Notes on refactoring/sharing the journey logic

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [ ] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/forms-web-app/src/dynamic-forms/questions.js [hunk 1]
```diff
@@ -29,44 +20,55 @@ const {
...
-	}
-} = require('../../src/config');
-const StringValidator = require('./validator/string-validator');
-const {
-	validation: {
+		},
 		characterLimits: { appealFormV2 }
 	}
-} = require('../config');
+} = require('../../src/config');
```
Comment (bryn-pins) [inline packages/forms-web-app/src/dynamic-forms/questions.js:26]: not sure what I did here

File: packages/forms-web-app/src/dynamic-forms/dynamic-components/date/question.js [hunk 2]
```diff
@@ -1,6 +1,31 @@
...
+];
+
+const formatDate = (date) => {
+	return {
+		date: date.getDate(),
+		day: weekdays[date.getDay()],
+		month: months[date.getMonth()],
+		year: date.getFullYear()
+	};
+};
```
Comment (beejjacobs) [inline packages/forms-web-app/src/dynamic-forms/dynamic-components/date/question.js:28]: This doesn't seem timezone-safe

Comment (bryn-pins) [reply packages/forms-web-app/src/dynamic-forms/dynamic-components/date/question.js:28]: yes

File: packages/forms-web-app/src/dynamic-forms/dynamic-components/utils/submission-information-utils.js [hunk 3]
```diff
@@ -15,6 +15,7 @@ const typeCodeToSubmissionInformationString = {
 exports.formatBeforeYouStartSection = async (appellantSubmission) => {
 	const { LPACode, appealTypeCode, applicationDecisionDate } = appellantSubmission;
 
+	// todo(journey-refactor): api call
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/dynamic-forms/dynamic-components/utils/submission-information-utils.js:18]: I think we could probably just take this file out and offer a way to push extra data into summaries?

File: packages/forms-web-app/src/dynamic-forms/validator/multifile-upload-validator.js [hunk 4]
```diff
@@ -1,6 +1,8 @@
 const BaseValidator = require('./base-validator.js');
 
 const multifileUploadSchema = require('../../validators/common/schemas/multifile-upload-schema.js');
+// todo(journey-refactor): uses clamav, config etc, inject this?
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/dynamic-forms/validator/multifile-upload-validator.js:4]: I think maybe we share some but not all of our validators? If it's a bring-your-own-questions type of affair then BO can plug in whatever they need at that point

File: packages/forms-web-app/src/dynamic-forms/validator/date-validator.js [hunk 5]
```diff
@@ -107,11 +107,13 @@ class DateValidator extends BaseValidator {
 					if (req.body[dayInput]) {
 						return this.noMonthErrorMessage;
 					}
+					// todo: empty error message bug
 				}),
 			body(yearInput)
 				.notEmpty()
 				.withMessage((_, { req }) => {
 					if (req.body[dayInput] && req.body[monthInput]) return this.noYearErrorMessage;
+					// todo: empty error message bug
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/dynamic-forms/validator/date-validator.js:116]: what's the bug?

File: packages/forms-web-app/src/dynamic-forms/validator/multifile-upload-validator.js (hunk 4 above)
Comment (bryn-pins) [reply packages/forms-web-app/src/dynamic-forms/validator/multifile-upload-validator.js:4]: yes makes sense, although thinking we don't share the entire multifile upload component

File: packages/forms-web-app/src/dynamic-forms/dynamic-components/utils/submission-information-utils.js (hunk 3 above)
Comment (bryn-pins) [reply packages/forms-web-app/src/dynamic-forms/dynamic-components/utils/submission-information-utils.js:18]: not sure we even need to share this right?

File: packages/forms-web-app/src/dynamic-forms/validator/date-validator.js (hunk 5 above)
Comment (bryn-pins) [reply packages/forms-web-app/src/dynamic-forms/validator/date-validator.js:116]: https://github.com/Planning-Inspectorate/appeal-planning-decision/pull/3802

Comment (bryn-pins) [reply packages/forms-web-app/src/dynamic-forms/validator/date-validator.js:116]: probably a better way to resolve this, but it validates each field separately, 
But we only want 1 error if each is left empty, not 3
It adds all but just returns undefined for the message for month/year, leaving an empty link in the page that can be accessed with keyboard

**Pull request closed without merging**